kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -191,9 +210,14 @@ private static void create8021xOptionalCaCertAndAnonIdentity(NetworkProperties p
             settings.put("anonymous-identity", new Variant<>(anonymousIdentity.get()));
         }
 
-        Optional<String> caCert = props.getOpt(String.class, "net.interface.%s.config.802-1x.ca-cert", deviceId);
-        if (caCert.isPresent()) {
-            settings.put("ca-cert", new Variant<>(caCert.get().getBytes(StandardCharsets.UTF_8)));
+        try {
+            Optional<Certificate> caCert = props.getOpt(Certificate.class,
+                    "net.interface.%s.config.802-1x.ca-cert-name", deviceId);
+            if (caCert.isPresent()) {
+                settings.put("ca-cert", new Variant<>(caCert.get().getEncoded()));
+            }
```

Comment:
Prefer the `ifPresent` syntax

```suggestion
            caCert.ifPresent(value -> {
                settings.put("ca-cert", new Variant<>(value.getEncoded()));
            });
```